service/history/mutableStateBuilder.go

```diff
@@ -1824,9 +1824,25 @@ func (e *mutableStateBuilder) ReplicateWorkflowExecutionStartedEvent(
 
 	e.executionInfo.Attempt = event.GetAttempt()
 	if !timestamp.TimeValue(event.GetWorkflowExecutionExpirationTime()).IsZero() {
-		e.executionInfo.RetryExpirationTime = event.GetWorkflowExecutionExpirationTime()
+		e.executionInfo.WorkflowExecutionExpirationTime = event.GetWorkflowExecutionExpirationTime()
 	}
 
+	var workflowRunTimeoutTime time.Time
+	workflowRunTimeoutDuration := timestamp.DurationValue(e.executionInfo.WorkflowRunTimeout)
+	if workflowRunTimeoutDuration == 0 {
```

Comment:
Let's remove if-else and add a comment that we need to set it when timeout is not infinite